core/src/mindustry/game/EventType.java

```diff
@@ -206,6 +206,17 @@ public TapEvent(Player player, Tile tile){
         }
     }
 
+    /** Called when a player chooses an option in menu. */
+    public static class MenuOptionChooseEvent{
+        public final Player player;
+        public final int option;
+
+        public MenuOptionChooseEvent(Player player, int option){
```

Comment:
There's some potential ambiguity here. As far as I can tell, there's no way to identify *which menu* was used, only which option was chosen. A server may have multiple menus.

`Call.menu` should accept a unique menu ID that is returned by the player once they choose something, to make sure options are handled for the right menu.

Reply:
I wouldn't agree with that.

Despite the fact that player can have multiple menus open at once, he can only interact with *one* and I think it is server's job to track which menu was last shown to player. It is also would be a bad UI design to show multiple menus to player from server's perspective, since they have a little transparency and hover over each other.

Reply:
Yes, multiple menus being shown at once aren't a good idea. However:

> I think it is server's job to track which menu was last shown to player

This seems like a lot of work, especially when a server may have any number of menus in it. You would have to write to a `ObjectMap<Player, Menu>` (or similar) every time a menu is opened.

In the event that a server *does* open multiple menus, everything would break in funny ways once the user started pressing buttons. Assuming that menus will always be singular doesn't seem very future-proof, and may lead to flaky behavior.